File: docs/guides/javascript/src/searchWithRuleContextBuyer.ts
import { algoliasearch } from 'algoliasearch';

import type { SearchParams } from 'algoliasearch';

const getBuyerAccountId = () => {
  return ''; // Implement your logic here
};

const client = algoliasearch('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

// get the buyer account information
const buyer = getBuyerAccountId();
const searchParams: SearchParams = {
  query: '<YOUR_SEARCH_QUERY>',
  ruleContexts: [buyer],
};

await client.searchSingleIndex({ indexName: 'indexName', searchParams: searchParams });

File: docs/guides/python/searchWithRuleContextBuyer.py
from algoliasearch.search.client import SearchClientSync

from algoliasearch.search.models.search_params import SearchParams


def _get_buyer_account_id() -> str:
    # Implement your logic here
    return ""


_client = SearchClientSync("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")


# get the buyer account information
buyer = _get_buyer_account_id()
search_params = SearchParams(
    query="<YOUR_SEARCH_QUERY>",
    rule_contexts=[buyer],
)

_client.search_single_index(
    index_name="<YOUR_INDEX_NAME>",
    search_params=search_params,
)
